jmclient/jmclient/configure.py

```diff
@@ -508,16 +551,30 @@ def get_irc_mchannels():
         return _get_irc_mchannels_old()
 
     SECTION_NAME += ':'
-    irc_sections = []
+    sections = []
     for s in jm_single().config.sections():
         if s.startswith(SECTION_NAME):
-            irc_sections.append(s)
-    assert irc_sections
+            sections.append(s)
+    assert sections
```

Comment:
Yes this seems pretty clear. I have rewritten it in this general form in 4cff6d7bb1e3f81473c99dccd3460aae2f2a6c55 . Factored out the parsing of irc and onion sections to sub-functions.
The new version just ignores whatever's after `MESSAGING:` .. so as suggested, the logic is: if type is onion, do onion, if type is irc, do irc, if type is not specified, do irc. But a slight problem with the test was that `onion1` was hardcoded where we read the `regtest_count` thing to figure out how many bots to start (i.e. you have to read it right at the start). Since that's just a test-specific thing, not much to worry about.

The main thing is that the refactored version works in test for now. Let me know if it looks OK.